src/visGeometry/ColorHandler.ts

```diff
@@ -0,0 +1,198 @@
+import { map, round, isEqual } from "lodash";
+import { Color } from "three";
+import SimulariumRenderer from "./rendering/SimulariumRenderer";
+
+export function convertColorStringToNumber(color: number | string): number {
+    if (typeof color !== "string") {
+        return color;
+    }
+    return parseInt(color.toString().replace(/^#/, "0x"), 16);
+}
+
+export function convertColorNumberToString(color: number | string): string {
+    if (typeof color === "string") {
+        return color;
+    }
+    return "#" + new Color(color).getHexString();
+}
+
+export const checkHexColor = (color: string): string => {
+    const hexColorCodeRegex = /^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/;
+    if (!hexColorCodeRegex.test(color)) {
+        throw new Error("Invalid color code");
+    }
+    return color;
+};
+
+class ColorHandler {
+    public idColorMapping: Map<number, number>;
+    private colorsData: Float32Array;
+    private renderer: SimulariumRenderer;
+
+    constructor(colors: (number | string)[], renderer: SimulariumRenderer) {
+        this.idColorMapping = new Map<number, number>();
+        this.colorsData = this.setColorArray(colors);
+        this.renderer = renderer;
```

Comment:
I wasn't really sure about storing the renderer. It does make referencing it easier, but I could also just send in the renderer to each method when their called. I tried moving everything that used the renderer into here, but there was one function that made that difficult. Let me know what you think

Reply:
I would say this class should know nothing about the renderer.  I would not have this class call any update functions on the renderer - it should just provide an accessor function to the raw data array and there are only one or two places in visGeometry/index.ts where you need to ask for that array and pass it to the renderer method.